R/diagnostics.R

```diff
@@ -35,19 +39,20 @@ epidist_diagnostics <- function(fit) {
   }
   if (fit$algorithm == "sampling") {
     np <- brms::nuts_params(fit)
-    divergent_indices <- np$Parameter == "divergent__"
-    treedepth_indices <- np$Parameter == "treedepth__"
-    diagnostics <- data.table(
-      "time" = sum(rstan::get_elapsed_time(fit$fit)),
-      "samples" = nrow(np) / length(unique(np$Parameter)),
-      "max_rhat" = round(max(brms::rhat(fit), na.rm = TRUE), 3),
-      "divergent_transitions" = sum(np[divergent_indices, ]$Value),
-      "per_divergent_transitions" = mean(np[divergent_indices, ]$Value),
-      "max_treedepth" = max(np[treedepth_indices, ]$Value)
-    )
-    diagnostics[, no_at_max_treedepth :=
-                  sum(np[treedepth_indices, ]$Value == max_treedepth)]
-    diagnostics[, per_at_max_treedepth := no_at_max_treedepth / samples]
+    divergent_ind <- np$Parameter == "divergent__"
+    treedepth_ind <- np$Parameter == "treedepth__"
+    diagnostics <- dplyr::tibble(
+      time = sum(rstan::get_elapsed_time(fit$fit)),
+      samples = nrow(np) / length(unique(np$Parameter)),
+      max_rhat = round(max(brms::rhat(fit), na.rm = TRUE), 3),
+      divergent_transitions = sum(np[divergent_ind, ]$Value),
+      per_divergent_transitions = mean(np[divergent_ind, ]$Value),
+      max_treedepth = max(np[treedepth_ind, ]$Value)
+    ) |>
```

Comment:
native pipe is a more stringent requirement on R version - is it necessary and / or already driven by another dependency?

Reply:
No don't think it's necessary, hadn't thought much about it thanks for noticing. Does using %>% introduce any further dependencies? If not then I guess we should switch

Reply:
To discuss here https://github.com/epinowcast/epidist/issues/336

Reply:
https://style.tidyverse.org/pipes.html